### DISCUSSION TITLE
Next Link not scrolling to hash

### DISCUSSION THREAD

**Discussion by ShadowRanger** [vercel/next.js#13134] [Help]
So I have a simple link that is meant to take you to a specific image on a page, however it does not seem to work. Instead, it just takes you to the very top of the page. I know it is not an issue with me adding the id property to the image component, as going directly to the href in the address bar works fine and scrolls me to the correct hash, it is just not working with the anchor.
```
<NextLink href={`/images#1`} passHref>
	<Button size="small" color="primary">
		View Image
	</Button>
</NextLink>
```
Any ideas? Thanks in advance.

**Comment by Florian-crg** [comment]
Maybe try something like this
```
<Link href="/images#a">
          <a>About Us</a>
 </Link>

<img src="/img.img" id="a" /> 
```

**Reply by ShadowRanger** [reply]
I could probably get it to work this way, but unfortunately in my case it needs to be rendered with that Button component.

**Reply by Florian-crg** [reply]
Where NextLink components comes from ? 
What is the link of the page without the anchor ? Local host:****/images ?

**Comment by jamesmosier** [comment]
Is `<Button>` a component or styled-component?

Also is the actual element that `<Button>` renders a `button`?

**Reply by ShadowRanger** [reply]
`<Button>` is a component from the Material UI library. It is currently rendering an anchor, not a button.

**Reply by jamesmosier** [reply]
I was able to get your scenario working 🤔 

I cloned [this example](https://github.com/mui-org/material-ui/tree/master/examples/nextjs)

Added this to the `pages/index.js`:

```jsx
<NextLink href="/about#foo" passHref>
  <Button size="small" color="primary">
    go to about#foo
  </Button>
</NextLink>
```

and then in `pages/about.js` i added 

```jsx
<div
  id="foo"
  style={{ marginTop: '1000px', height: '1000px', background: 'red' }}
>
  foo
</div>
```

and when i clicked the link on the index page it would take me to `/about#foo` and scroll me down to the bottom where this element lives.

**Reply by rehanpunjwani** [reply]
This doesn't work until we refresh the /about#foo. Correct me if I am wrong!

**Comment by helloworld** [comment]
This sounds similar to the issue I was having. Here's the fix in our codebase that worked for us: https://dagster.phacility.com/source/dagster/browse/master/docs/next/src/pages/_app.tsx$31-46

**Reply by finorix079** [reply]
It is deleted.

**Comment by andrewmundy** [comment]
@helloworld 's `setTimeout` solution

this is the correct link

https://dagster.phacility.com/source/dagster/browse/master/docs/next/src/pages/_app.tsx;cf558f64e96a77a8f9afb3f40eed4b579b8a171a

**Reply by yarastqt** [reply]
I guess it's bad solution, because this breaks history back case :(
